src/download.rs

```diff
@@ -18,58 +23,141 @@ struct LatestReleaseApiResponse {
+            };
+            let architecture = match std::env::consts::ARCH {
+                "aarch64" => "arm64",
+                "x86_64" => "amd64",
+                unsupported_arch => bail!("Unsupported architecture: {}", unsupported_arch),
+            };
+
+            Ok(format!("forc-binaries-{}_{}.tar.gz", os, architecture))
+        }
 
-    Ok(format!(
-        "fuel-core-{}-{}-{}-{}.tar.gz",
-        // strip the 'v' from the version string to match the file name of the releases
-        &version[1..version.len()],
-        architecture,
-        vendor,
-        os
-    ))
+        component::FUEL_CORE => {
+            let architecture = match std::env::consts::ARCH {
+                "aarch64" | "x86_64" => std::env::consts::ARCH,
+                unsupported_arch => bail!("Unsupported architecture: {}", unsupported_arch),
+            };
+
+            let vendor = match std::env::consts::OS {
+                "macos" => "apple",
+                _ => "unknown",
+            };
+
+            let os = match std::env::consts::OS {
+                "macos" => "darwin",
+                "linux" => "linux-gnu",
+                unsupported_os => bail!("Unsupported os: {}", unsupported_os),
+            };
+
+            Ok(format!(
+                "fuel-core-{}-{}-{}-{}.tar.gz",
+                &download_cfg.version.to_string(),
+                architecture,
+                vendor,
+                os
+            ))
+        }
+        component::FUELUP => {
+            let architecture = match std::env::consts::ARCH {
+                "aarch64" | "x86_64" => std::env::consts::ARCH,
+                unsupported_arch => bail!("Unsupported architecture: {}", unsupported_arch),
+            };
+
+            let vendor = match std::env::consts::OS {
+                "macos" => "apple",
+                _ => "unknown",
+            };
+
+            let os = match std::env::consts::OS {
+                "macos" => "darwin",
+                "linux" => "linux-gnu",
+                unsupported_os => bail!("Unsupported os: {}", unsupported_os),
+            };
+
+            Ok(format!(
+                "fuelup-{}-{}-{}-{}.tar.gz",
+                &download_cfg.version.to_string(),
+                architecture,
+                vendor,
+                os
+            ))
+        }
+        _ => bail!("Unrecognized component: {}", download_cfg.name),
+    }
 }
 
-pub fn get_latest_tag(github_api_url: &str) -> Result<String> {
+pub fn get_latest_tag(github_api_url: &str) -> Result<Version> {
     let handle = ureq::builder().user_agent("fuelup").build();
     let resp = handle.get(github_api_url).call()?;
 
     let mut data = Vec::new();
     resp.into_reader().read_to_end(&mut data)?;
 
     let response: LatestReleaseApiResponse = serde_json::from_str(&String::from_utf8_lossy(&data))?;
-    Ok(response.tag_name)
+
+    let version = Version::parse(&response.tag_name[1..response.tag_name.len()])?;
```

Comment:
```suggestion
    // Given a semver version with preceding 'v' (e.g. `v1.2.3`), take the slice after 'v' (e.g. `1.2.3`).
    let version_str = &response.tag_name['v'.len()..];
    let version = Version::parse(version_str)?;
```

Reply:
Yep, this sounds good, i'll remove all the changes related to specific versioning from this PR for now, and we'll just merge changes related to the refactor